XenAdmin/Wizards/RollingUpgradeWizard/RollingUpgradeWizardPrecheckPage.cs

```diff
@@ -142,22 +142,32 @@ protected override List<CheckGroup> GenerateChecks(Pool_patch patch)
         {
             var groups = new List<CheckGroup>();
 
+            List<Host> hostsToUpgrade = new List<Host>();
+            List<Host> hostsToUpgradeOrUpdate = new List<Host>();
+            foreach (var pool in SelectedPools)
+            {
+                var poolHostsToUpgrade = pool.HostsToUpgrade();
+                hostsToUpgrade.AddRange(poolHostsToUpgrade);
+                hostsToUpgradeOrUpdate.AddRange(ApplyUpdatesToNewVersion
+                    ? HostsToUpgradeOrUpdate(pool)
+                    : poolHostsToUpgrade);
+            }
+
             //XenCenter version check (if any of the selected server version is not the latest)
             var latestCrVersion = Updates.XenServerVersions.FindAll(item => item.LatestCr).OrderByDescending(v => v.Version).FirstOrDefault();
             if (latestCrVersion != null &&
-                SelectedServers.Any(host => new Version(Helpers.HostProductVersion(host)) < latestCrVersion.Version))
+                hostsToUpgradeOrUpdate.Any(host => new Version(Helpers.HostProductVersion(host)) < latestCrVersion.Version))
             {
                 groups.Add(new CheckGroup(Messages.CHECKING_XENCENTER_VERSION, new List<Check> {new XenCenterVersionCheck(null)}));
             }
 
-            //HostMaintenanceModeCheck checks
+            //HostMaintenanceModeCheck checks - for hosts that will be upgraded or updated
             var livenessChecks = new List<Check>();
-            foreach (Host host in SelectedServers)
+            foreach (Host host in hostsToUpgradeOrUpdate)

```

Comment:
Can it be a problem that now the check for maintenance mode will be run on the hosts to update too? Previously for those hosts we only checked for liveness.

Reply:
I will change it so it doesn't check the maintenance mode for th ehosts that only need updated